GiniVision/Classes/ToolTipView.swift

```diff
@@ -0,0 +1,300 @@
+//
+//  ToolTipView.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 9/21/17.
+//
+
+import Foundation
+import UIKit
+
+final class ToolTipView: UIView {
+    
+    enum ToolTipPosition {
+        case above
+        case below
+        case left
+        case right
+    }
+    
+    fileprivate var arrowWidth:CGFloat = 20
+    fileprivate var arrowHeight:CGFloat = 20
+    fileprivate var closeButtonWidth:CGFloat = 20
+    fileprivate var closeButtonHeight:CGFloat = 20
+    fileprivate var itemSeparation: CGFloat = 16
+    fileprivate var margin:(top:CGFloat, left:CGFloat, right: CGFloat, bottom: CGFloat) = (20, 20, 20, 20)
+    fileprivate var maxWidthOnIpad:CGFloat = 375
+    fileprivate var padding:(top:CGFloat, left:CGFloat, right: CGFloat, bottom: CGFloat) = (16, 16, 16, 16)
+    
+    fileprivate var textWidth:CGFloat {
+        guard let superview = superview else { return 0 }
+        let maxWidth = UIDevice.current.isIpad ? maxWidthOnIpad : superview.frame.width
```

Comment:
I'd love it if a property like `textWidth` doesn't depend on UIDevice and what device the code is running on. Can't you use the superview's frame? Or not have `textWidth` altogether?

Reply:
I came up with an idea. Thanks! 👍